pkg/segment/query/processor/fillnullcommand.go

```diff
@@ -18,16 +18,87 @@
 package processor
 
 import (
+	"io"
+
 	"github.com/siglens/siglens/pkg/segment/query/iqr"
 	"github.com/siglens/siglens/pkg/segment/structs"
+	"github.com/siglens/siglens/pkg/segment/utils"
+	toputils "github.com/siglens/siglens/pkg/utils"
+	log "github.com/sirupsen/logrus"
 )
 
 type fillnullProcessor struct {
 	options *structs.FillNullExpr
 }
 
+func getTheFinalFillValue(fillValue string, qid uint64) (utils.CValueEnclosure, error) {
+	fillValueDtype, err := utils.CreateDtypeEnclosure(fillValue, qid)
+	if err != nil {
+		return utils.CValueEnclosure{}, toputils.TeeErrorf("qid=%v, performFillNullForTheFields: cannot create dtype for the fill Value; err=%v", qid, err)
+	}
+
+	var finalFillValue interface{}
+
+	if fillValueDtype.IsBool() {
+		finalFillValue = fillValueDtype.BoolVal
+	} else if fillValueDtype.IsInt() {
+		finalFillValue = fillValueDtype.SignedVal
+	} else if fillValueDtype.IsFloat() {
+		finalFillValue = fillValueDtype.FloatVal
+	} else {
+		finalFillValue = fillValue
+	}
+
+	return utils.CValueEnclosure{CVal: finalFillValue, Dtype: fillValueDtype.Dtype}, nil
+}
+
+func performFillNullForTheFields(iqr *iqr.IQR, fields []string, cTypeFillValue utils.CValueEnclosure) {
+	fillNullForAllRecords := make(map[string]struct{})
+
+	for _, field := range fields {
+		values, err := iqr.ReadColumn(field)
+		if err != nil {
+			fillNullForAllRecords[field] = struct{}{}
+			continue
+		}
```

Comment:
nit: I'm not really sure why you're adding this to a map to handle slightly later instead of handling this column right now. But if you like this way better, feel free to keep it